Header header
node_frame[] node_frames
int32 self_id